src/components/Modal/Modal.tsx

```diff
@@ -0,0 +1,113 @@
+import * as React from 'react';
+import { TPAComponentsConsumer } from '../TPAComponentsConfig';
+
+import styles from './Modal.st.css';
+import { IconButton } from '../IconButton';
+import { Close } from '../../assets/icons';
+import { DATA_HOOKS } from './constants';
+
+export interface ModalProps {
+  /**
+   * Describing if the modal should be shown or not.
+   */
+  isOpen: boolean;
+
+  /**
+   * Function that will be run when the modal is requested to be closed (by clicking on overlay)
+   * Note: It is not called if isOpen is changed by other means.
+   */
+  onRequestClose(): void;
+
+  /**
+   * Describing if the modal should be shown in full screen or not.
+   */
+  inFullScreen?: boolean;
+
+  /**
+   * Describing if the modal should be shown with close button or not.
+   */
+  withCloseButton?: boolean;
+
+  /**
+   * Describing if the modal should be shown with dark background overlay or not.
+   */
+  withBackground?: boolean;
+  aria?: {
+    labeledBy?: string;
+  };
+}
+
+interface DefaultProps {
+  inFullScreen: boolean;
+  withCloseButton: boolean;
+  withBackground: boolean;
+  aria: {
+    labeledBy: string;
+  };
+}
+
+interface State {}
+
+/** The modal component provides a solid foundation for creating dialogs, popovers, lightboxes, or whatever else. */
+export class Modal extends React.Component<ModalProps, State> {
+  static displayName = 'Modal';
+  static defaultProps: DefaultProps = {
+    inFullScreen: false,
+    withCloseButton: true,
+    withBackground: true,
```

Comment:
Even though it makes sense to have these two props true by default, I think it's weird to have it a `with...` prop be true without actually passing the prop.
So i'd either change the name to: `hideCloseButton`, or remove this default value.
(For both `withCloseButton` and `withBackground`)